src/services/projects.js

```diff
@@ -318,4 +321,280 @@ module.exports = class ProjectsHelper {
 			throw error
 		}
 	}
+
+	static async submitForReview(resourceId, bodyData, userDetails) {
+		try {
+			let projectDetails = await this.details(resourceId, userDetails.organization_id, userDetails.id)
+
+			if (projectDetails.statusCode == httpStatusCode.ok) {
+				let projectData = projectDetails.result
+				if (projectData.user_id !== userDetails.id) {
+					return responses.failureResponse({
+						message: 'DONT_HAVE_PROJECT_ACCESS',
+						statusCode: httpStatusCode.bad_request,
+						responseCode: 'CLIENT_ERROR',
+					})
+				}
+
+				let entityTypes = await entityModelMappingQuery.findEntityTypes(
+					{
+						model: common.PROJECT,
+						status: common.STATUS_ACTIVE,
+					},
+					['value', 'has_entities', 'validations']
+				)
+				const entities = []
+
+				for (let i = 0; i < entityTypes.length; i++) {
+					const entityType = entityTypes[i]
+
+					if (entityType.has_entities) {
+						const projectEntityData = projectData[entityType.value]
+
+						if (projectEntityData) {
+							if (Array.isArray(projectEntityData)) {
+								// If projectEntityData is an array, extract ids from each item
+								projectEntityData.forEach((item) => {
+									entities.push(item.value)
+								})
+							} else if (typeof projectEntityData === 'object') {
+								// If projectEntityData is an object, extract the id
+								entities.push(projectEntityData.value)
+							}
+						}
+					}
+				}
+
+				const allEntities = await entityService.read({ value: entities }, userDetails.id)
+
+				for (let i = 0; i < entityTypes.length; i++) {
```

Comment:
@ankit1shahu instead of i you can name something like pointerToEntityType or 
use for (const entityType of entityTypes)